Server/src/routes/DatasetRouter.ts

```diff
@@ -0,0 +1,52 @@
+import { Request, Response, Router } from 'express';
+import { DataSetController } from '../controllers/DataSetController';
+import { JWTAuthenticator } from '../middleware/JWTAuthenticator';
+
+/**
+ * This file contains the route for a call to query or obtain one or more data sets. 
+ * If an API call is made to /dataset* then the request is routed to the getDataController
+ * to continue processing of the request.
+ */
+
+let router = Router();
+let dataSetController = new DataSetController();
+
+//Note to Self: Verify which of these routes are protected 
+// JWTAuthenticator.verifyJWT, JWTAuthenticator.verifyAdmin
+router.get('/api/v1/dataset/userUploadedDatasets/:userUploadedDatasets', (request: Request, response: Response) => {
+    dataSetController.createRequestForUserUploadedDatasets(request, response);
+});
+
+router.get('/api/v1/dataset/userSavedDatsets/:userSavedDatsets', (request: Request, response: Response) => {
+    dataSetController.createRequestForUserSavedDatsets(request, response);
+});
+
+router.get('/api/v1/dataset/fetchUnapprovedDatasets$', (request: Request, response: Response) => {
+    dataSetController.createRequestForUnapprovedDatsets(request, response);
+});
+
+router.get('/api/v1/dataset*', (request: Request, response: Response) => {
+    dataSetController.createRequestForData(request, response);
+});
+
+router.delete('/api/v1/dataset/:datasetId', [JWTAuthenticator.verifyJWT, JWTAuthenticator.verifyAdmin], (request: Request, response: Response) => {
+    dataSetController.createRequestToRejectDataset(request, response)
+})
+
+router.put('/api/v1/flagDataSet/:datasetId', [JWTAuthenticator.verifyJWT, JWTAuthenticator.verifyAdmin], (request: Request, response: Response) => {

```

Comment:
a more conventional route would be api/v1/dataset/:datsetId/flag and it owuld be a parch method, because flag updates entity completely (all fields are updated so you must pass a new dataset in the request.body), patch updates entity partionally (one field for example)

Reply:
This route does not contain the dataset. The dataset is only passed on edit or upload.